common/lib/xmodule/xmodule/video_module/video_module.py

```diff
@@ -208,7 +209,24 @@ def prioritize_hls(self, youtube_streams, html5_sources):
 
         return False
 
-    def get_html(self):
+    def public_view(self, context):
+        """
+        Returns a fragment that contains the html for the preview view
+        """
+        return self.student_view(context, PUBLIC_VIEW)
+
+    def student_view(self, context, view=STUDENT_VIEW):  # pylint: disable=arguments-differ
```

Comment:
You can specify the default `view=STUDENT_VIEW` in the `get_html()` parameters. Then this method will not be needed as the implementation in the super class `XModule` will be sufficient.

Reply:
The `student_view` method in the super class only has context as a parameter (no view parameter). Hence I've overridden that here.

Reply:
Yes, but then you can change `public_view` body to `return Fragment(self.get_html(PUBLIC_VIEW))` instead of proxying it through `student_view`. Also we don't need the docs when just overridding from a superclass without change in behavior. So we can remove that.